src/presenter/details.js

```diff
@@ -0,0 +1,88 @@
+define(["../model/stops", "../model/departures", "../view/details"],
+function (stops, departures, view) {
+    console.log("Details module loaded");
+
+    // It allows us to navigate through the history without request it again
+    var lastHash = "";
+
+    // In case of the user back to the search view before the result has come
+    var canceledFetch = false;
+
+    // Callback executed when the AJAX finishes
+    function successStops() {
+        if (canceledFetch) {
+            return;
+        }
+        var values = stops.getStops();
+        view.renderStops(values);
+        view.appendStops();
+    }
+
+    // Callback executed when the AJAX finishes
+    function successDepartures() {
+        if (canceledFetch) {
+            return;
+        }
+        var values = departures.getDepartures();
+        view.renderDepartures(values);
+        view.appendDepartures();
+    }
+
+    // Loads details (stops and departures) fetching them by id
+    function load(id) {
+        console.log("[details] Load id: " + id);
+        canceledFetch = false;
+        stops.fetch(id, successStops);
+        departures.fetch(id, successDepartures);
+    }
+
+    // When back button is pressed
+    view.on("click", function () {
+        canceledFetch = true;
+        window.history.back();
+    });
+
+    function navigate() {
```

Comment:
It's not, but would be a nice to have if you think about the user
experience.

--

Ismael Stahelin, Software Engineer
ArcTouch - App Development Studio

ismael.stahelin@arctouch.com
+1-415-944-2000 main

twitter.com/arctouch

Custom apps for world-class brands and the Fortune 500
arctouch.com/work | arctouch.com/blog

On Mon, Oct 5, 2015 at 3:52 PM, Diego Weiss <notifications@github.com>
wrote:

> In src/presenter/details.js
> <https://github.com/diegolw/floripaByBus/pull/1#discussion_r41208136>:
>
> > +
> > +    // Loads details (stops and departures) fetching them by id
> > +    function load(id) {
> > +        console.log("[details] Load id: " + id);
> > +        canceledFetch = false;
> > +        stops.fetch(id, successStops);
> > +        departures.fetch(id, successDepartures);
> > +    }
> > +
> > +    // When back button is pressed
> > +    view.on("click", function () {
> > +        canceledFetch = true;
> > +        window.history.back();
> > +    });
> > +
> > +    function navigate() {
>
> I have saved only the last search. If you navigate through forward and
> backward button you can see that the request is not triggered again. I
> preserve the state in the HTML elements. I didn't save some request because
> I thought that was not a requirement. My fault!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/diegolw/floripaByBus/pull/1/files#r41208136>.
>